gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch

```diff
@@ -0,0 +1,2123 @@
++  return;
++
++wrong_state:
++  {
++    GST_WARNING_OBJECT (encoder, "setting property in wrong state");
++    GST_OBJECT_UNLOCK (encoder);
++  }
++}
++
++static void
++gst_svthevc_enc_get_property (GObject * object, guint prop_id,
++    GValue * value, GParamSpec * pspec)
++{
++  GstSvtHevcEnc *encoder;
++
++  encoder = GST_SVTHEVC_ENC (object);
++
++  GST_OBJECT_LOCK (encoder);
++  switch (prop_id) {
++    case PROP_INSERT_VUI:
++      g_value_set_boolean (value, encoder->insert_vui);
++      break;
++    case PROP_AUD:
++      g_value_set_boolean (value, encoder->aud);
++      break;
++    case PROP_HIERARCHICAL_LEVEL:
++      g_value_set_uint (value, encoder->hierarchical_level);
++      break;
++    case PROP_LOOKAHEAD_DISTANCE:
++      g_value_set_uint (value, encoder->la_depth);
++      break;
++    case PROP_ENCODER_MODE:
++      g_value_set_uint (value, encoder->enc_mode);
++      break;
++    case PROP_RC_MODE:
++      g_value_set_uint (value, encoder->rc_mode);
++      break;
++    case PROP_QP:
++      g_value_set_uint (value, encoder->qp);
++      break;
++    case PROP_QP_MAX:
++      g_value_set_uint (value, encoder->qp_max);
++      break;
++    case PROP_QP_MIN:
++      g_value_set_uint (value, encoder->qp_min);
++      break;
++    case PROP_SCENE_CHANGE_DETECTION:
++      g_value_set_boolean (value, encoder->scene_change_detection);
++      break;
++    case PROP_TUNE:
++      g_value_set_uint (value, encoder->tune);
++      break;
++    case PROP_LATENCY_MODE:
++      g_value_set_uint (value, encoder->latency_mode);
++      break;
++    case PROP_BASE_LAYER_SWITCH_MODE:
++      g_value_set_uint (value, encoder->base_layer_switch_mode);
++      break;
++    case PROP_BITRATE:
++      g_value_set_uint (value, encoder->bitrate);
++      break;
++    case PROP_KEY_INT_MAX:
++      g_value_set_int (value, encoder->keyintmax);
++      break;
++    case PROP_INTRA_REFRESH:
++      g_value_set_uint (value, encoder->intra_refresh);
++      break;
++    case PROP_CORES:
++      g_value_set_uint (value, encoder->cores);
++      break;
++    case PROP_SOCKET:
++      g_value_set_int (value, encoder->socket);
++      break;
++    case PROP_TILE_ROW:
++      g_value_set_uint (value, encoder->tile_row);
++      break;
++    case PROP_TILE_COL:
++      g_value_set_uint (value, encoder->tile_col);
++      break;
++    default:
++      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
++      break;
++  }
++  GST_OBJECT_UNLOCK (encoder);
++}
++
++static gboolean
++plugin_init (GstPlugin * plugin)
++{
++  GST_DEBUG_CATEGORY_INIT (svthevc_enc_debug, "svthevcenc", 0,
++      "h265 encoding element");
++
++  return gst_element_register (plugin, "svthevcenc",
++      GST_RANK_PRIMARY, GST_TYPE_SVTHEVC_ENC);
++}
++
++GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
++    GST_VERSION_MINOR,
++    svthevcenc,
++    "svt-hevc encoder based H265 plugins",
++    plugin_init, VERSION, "GPL", GST_PACKAGE_NAME, GST_PACKAGE_ORIGIN)
```

Comment:
Could you update the plugin info?

#ifndef VERSION
#define VERSION "1.0"
#endif
#ifndef PACKAGE
#define PACKAGE "gstreamer-svt-hevc"
#endif
#ifndef PACKAGE_NAME
#define PACKAGE_NAME "SVT-HEVC Encoder plugin for GStreamer"
#endif
#ifndef GST_PACKAGE_ORIGIN
#define GST_PACKAGE_ORIGIN "https://github.com/OpenVisualCloud"
#endif

GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
    GST_VERSION_MINOR,
    svthevcenc,
    "Scalable Video Technology for HEVC Encoder (SVT-HEVC Encoder)",
    plugin_init, VERSION, "GPL", PACKAGE_NAME, GST_PACKAGE_ORIGIN)

Reply:
I think the PACKAGE_NAME and PACKAGE_ORIGIN are not the information of SVT-HEVC encoder plugin, it seems to be the information of the package to which the plugin belongs. (e.g. gst-pluings-bad). What do you think?

Reply:
Agree.